switchboard/manager.py

```diff
@@ -83,12 +85,37 @@ def __getitem__(self, key):
         """
         return SwitchProxy(self, super(SwitchManager, self).__getitem__(key))
 
+    def with_result_cache(func):
+        """
+        Decorator specifically for is_active.  If self.result_cache is set to a {}
+        the is_active results will be cached for each set of params.
```

Comment:
Yep, you got it.  It isn't completely generic (can't be applied to other functions) since it stores results in the `self.result_cache` rather than a hidden field like most generic memoize decorators.  This is so `self.result_cache` is an easily accessible for clearing.